parser/src/lexer.rs

```diff
@@ -655,6 +655,12 @@ where
             // Detect indentation levels
             if self.at_begin_of_line {
                 self.handle_indentations()?;
+                if self.mode == Mode::Jupyter
+                    // https://github.com/ipython/ipython/blob/635815e8f1ded5b764d66cacc80bbe25e9e2587f/IPython/core/inputtransformer2.py#L345
+                    && matches!(self.window[0], Some('%' | '!' | '?' | '/' | ';' | ','))
+                {
+                    self.lex_and_emit_magic_command();
+                }
```

Comment:
This move is done to handle and emit indentation tokens, earlier we would consume it as the entire line containing the line magic was ignored by the parser.